share/wake/lib/system/remote_cache_api.wake

```diff
@@ -24,6 +27,123 @@ export tuple RemoteCacheApi =
     # The api key required for authenicated routes. Not required for all requests
     export Authorization: Option String
 
+## --- POST /job/matching types ---
+
+# A request to the remote server to check if a job is cached
+tuple CacheSearchRequest =
+    # The command line of the job
+    Cmd: List String
+    # The directory of the job
+    Cwd: String
+    # The environment of the job
+    Env: List String
+    # User provided string to mark two jobs as different even if wake considers them the same
+    HiddenInfo: String
+    # If the job was ran as if a user was running it from a terminal
+    IsAtty: Boolean
+    # The stdin of the job
+    Stdin: String
+    # The list of files the job is allowed to read in the sandbox
+    VisibleFiles: List Path
+
+# A downloadable blob returned from the remote shared cache
+tuple CacheSearchBlob =
+    # The id of the blob on the remote server
+    Id: String
+    # The uri where the blob may be downloaded from. Not guaranteed to be http/https
+    Uri: String
+
+# A single file backed by a blob that a cached job output
+tuple CacheSearchOutputFile =
+    # The path on disk of the file
+    Path: String
+    # The mode on disk of the file
+    Mode: Integer
+    # The description of how to download the files contents
+    Blob: CacheSearchBlob
+
+# A match response from the server. Specifies all details of a cached job
+tuple CacheSearchResponseMatch =
+    # The symlinks that were output by the job
+    OutputSymlinks: List String
+    # The directories that were output by the job
+    OutputDirs: List String
+    # The files that were output by the job
+    OutputFiles: List CacheSearchOutputFile
+    # The stdout of the job
+    StdoutBlob: CacheSearchBlob
+    # The stderr of the job
+    StderrBlob: CacheSearchBlob
+    # The code that the job return on exit
+    Status: Integer
+    # The total runtime of the job
+    Runtime: Double
+    # The total cputime of the job
+    Cputime: Double
+    # The total memory of the job
+    Memory: Integer
+    # The number of bytes read by the job
+    IBytes: Integer
+    # The number of bytes written by the job
+    OBytes: Integer
```

Comment:
More specifically any type that is interfacing with an external system or dependancy should be a POD (https://en.wikipedia.org/wiki/Passive_data_structure) so that it can easily be replaced/side stepped/updated as the external system changes

Reply:
Cool. Thanks for the explanation!